simpleclient_hotspot/src/main/java/io/prometheus/client/hotspot/StandardExports.java

```diff
@@ -101,6 +98,49 @@ public StandardExports() {
     return mfs;
   }
 
+  static Long callLongGetter(String getterName, Object obj) throws NoSuchMethodException {
+    return callLongGetter(obj.getClass().getMethod(getterName), obj);
+  }
+
+  /**
+   * Attempts to call a method either directly or via one of the implemented interfaces.
+   * <p>
+   * A Method object refers to a specific method declared in a specific class. The first invocation
+   * might happen with method === SomeConcreteClass.publicLongGetter() and will fail if
+   * SomeConcreteClass is not public. We then recurse over all interfaces implemented by
+   * SomeConcreteClass (or extended by those interfaces and so on) until we eventually invoke
+   * callMethod() with method === SomePublicInterface.publicLongGetter(), which will then succeed.
+   * <p>
+   * There is a built-in assumption that the method will never return null (or, equivalently, that
+   * it returns the primitive data type, i.e. {@code long} rather than {@code Long}). Similarly,
+   * there is an assumption that the method doesn't throw an exception. If one of these assumptions
+   * doesn't hold, the method might be called repeatedly and the returned value will be null (any
+   * exception will be silently swallowed).
+   */
+  static Long callLongGetter(Method method, Object obj) {
+    try {
+      return (Long) method.invoke(obj);
+    } catch (Exception e) {
```

Comment:
Replaced with IllegalAccessException and am now throwing InvocationTargetException when the invoked method throws, thus aborting the recursion if that happens.